### PR TITLE
Bug 1868061: switch watchtools.UntilWithoutRetry to watchtools.UntilWithSync in deployment tests

### PR DESCRIPTION
**Pull request by soltysh**
/assign @tnozicka

### CODE REVIEW

**Comment by soltysh** [comment]
@tnozicka looks like one more `UntilWithoutRetry` was still left after your #25010, ptal

File: test/extended/deployments/util.go [hunk 1]
```diff
@@ -405,15 +408,37 @@ func waitForSyncedConfig(oc *exutil.CLI, name string, timeout time.Duration) err
...
+			}, name)
+		}
+		return false, nil
+	}
+
 	var rc *corev1.ReplicationController
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
-	if _, err := watchtools.UntilWithoutRetry(ctx, watcher, func(e watch.Event) (bool, error) {
+	if _, err := watchtools.UntilWithSync(ctx, rcListWatch, &core.ReplicationController{}, preconditionFunc, func(e watch.Event) (bool, error) {
```
Comment (tnozicka) [inline test/extended/deployments/util.go:441]: waitForRCState

**Review by tnozicka — APPROVED** [review]
/lgtm

**Pull request merged**